### ISSUE TITLE
EQSANS transmissions fitting errors: use Mantid's errors instead of generating the error ourselves

### ISSUE THREAD

**Issue by glass-ships** [neutrons/drtsans#499]
After the raw transmission is fitted, we calculate the errors for the fitted transmission in `drtsans.tof.eqsans.transmission.insert_fitted_errors`

```
def insert_fitted_errors(input_workspace, mantid_fit_output):
    r"""
    Substitute raw errors with errors derived from the model transmission
    over a wavelength band. Substitute with zero errors elsewhere

    Errors are calculated using the errors in the fitting parameters of the transmission model.
    For instance, the errors in the slope and intercept of a linear model.
    We employ numerical derivative of the fit function with respect to the fitting parameters

    Parameters
    ----------
    input_workspace: str, ~mantid.api.MatrixWorkspace
        Workspace to contain the fitted error transmission values
    mantid_fit_output: namedtuple
        Return value of Mantid's Fit algorithm
    lower_wavelength_boundary: float
        Lower wavelength boundary of the range of fitted transmission values
    upper_wavelength_boundary: float
        Upper wavelength boundary of the range of fitted transmission values
    """
    fit_function = mantid_fit_output.Function  # Fit function object

    # Find the fitting wavelengths, and use the midpoint wavelengths for the error estimation
    fitting_wavelength_range = mantid_fit_output.OutputWorkspace.readX(0)
    fitting_wavelengths = (fitting_wavelength_range[: -1] + fitting_wavelength_range[1:]) / 2  # midpoints

    # Iterate over the fitting parameters, calculating their numerical derivatives and their error contributions
    fitting_errors = np.zeros(len(fitting_wavelengths))  # we accumulate the individual errors in this array
    parameter_table = mantid_fit_output.OutputParameters  # contain the optimized parameters of the
    for row_index in range(parameter_table.rowCount() - 1):  # last row is the Chi-square, thus we exclude it
        row = parameter_table.row(row_index)
        parameter_name, parameter_value, parameter_error = [row[key] for key in ('Name', 'Value', 'Error')]
        # evaluate the fit function by increasing the parameter value
        fit_function[parameter_name] += parameter_error  # slightly increase the parameter's value
        evaluation_plus = fit_function(fitting_wavelengths)  # evaluate function at the fitting wavelengths
        # re-evaluate the fit function by decreasing the parameter value
        fit_function[parameter_name] -= 2 * parameter_error
        evaluation_minus = fit_function(fitting_wavelengths)
        numerical_derivative = (evaluation_plus - evaluation_minus) / (2 * parameter_error)
        fitting_errors += (numerical_derivative * parameter_error)**2  # error contribution from this parameter
        fit_function[parameter_name] += parameter_error  # restore the original value
    fitting_errors = np.sqrt(fitting_errors)

    # Find the array indexes enclosing the range of fitted wavelengths
    first_fitted_wavelength, last_fitted_wavelength = fitting_wavelength_range[0], fitting_wavelength_range[-1]
    input_handle = mtd[str(input_workspace)]
    input_wavelength_range = input_handle.readX(0)
    first_insertion_index = np.where(input_wavelength_range == first_fitted_wavelength)[0][0]
    last_insertion_index = np.where(input_wavelength_range == last_fitted_wavelength)[0][0]

    # Insert the fitting errors, and set zero elsewhere.
    input_handle.dataE(0)[:] = np.zeros(input_handle.dataE(0).size)
    input_handle.dataE(0)[first_insertion_index: last_insertion_index] = fitting_errors
```

This function assumes the fitting parameters are **uncorrelated** and calculates the error in the fitted transmission `y(x)` with the typical error propagation formula.

For a linear model `y(x) = a + b * x`, the error in the fit becomes `y_err^2 = C_aa  + x^2 * C_bb`, where C_aa and C_bb are the diagonal elements of the covariance matrix for the fit parameters. These elements are just the square of the errors for the fitting parameters `a` and `b`.

Mantid uses the fitting library <a href="https://www.gnu.org/software/gsl/doc/html/">gsl</a> which estimates the error without the assumption that the fitting parameters are uncorrelated. For a linear model, gsl uses function <a href="https://www.gnu.org/software/gsl/doc/html/lls.html#c.gsl_fit_linear_est">gsl_fit_linear_est</a> to estimate `y_err` as `y_err^2 = C_aa  + x^2 * C_bb + 2 * x * C_ab` where `C_ab` is the non-diagonal term of the covariance matrix for the fit parameters.

Here's a fit session in Mantid for some data that roughly follows the line `y(x) = 3 + 2 * x`. We can use the fitting errors and the normalized covariance matrix reported by the Mantid fit session in order to calculate `my_errors^2 = C_aa  + x^2 * C_bb` and `gsl_errors^2 = C_aa  + x^2 * C_bb + 2 * x * C_ab`

<img src="/uploads/7dad91c0d4a1a818d0fb89d613199a03/Selection_018.png" width="750">

In this fitting session, `C_aa = 0.6698^2`, `C_bb = 0.1291^2` and `C_ab = 0.6698 * 0.1291 * (-0.8674)`

`gsl_errors` are smaller than `my_errors` because `C_ab` is negative (anti-correlation)

In summary, we're better of using the errors provided by Mantid as they include correlations of fitting parameters.

**Comment by glass-ships** [comment]
This issue was migrated from [code.ornl.gov](https://code.ornl.gov/sns-hfir-scse/sans/sans-backend/-/issues/453).
